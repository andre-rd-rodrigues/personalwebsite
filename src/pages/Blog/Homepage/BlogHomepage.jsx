import AnimatedHeading from "components/AnimatedHeading/AnimatedHeading";
import ArticlesGrid from "components/Blog/ArticlesGrid/ArticlesGrid";
import TopArticle from "components/Blog/TopArticle/TopArticle";
import PageContainer from "components/PageContainer/PageContainer";
import { blog } from "mocks/data";
import React from "react";
import styles from "./bloghomepage.module.scss";
import {
  blogHomepageDelayVariant,
  fadeInVariant,
  motion
} from "assets/motion/motionVariants";

function BlogHomepage() {
  return (
    <PageContainer
      color="dark"
      bgColor="grey-yellow"
      className={styles.container}
    >
      <div className={styles.blogHomepage}>
        <AnimatedHeading>
          <h1>ideas & thoughts</h1>
        </AnimatedHeading>
        <motion.div
          variants={blogHomepageDelayVariant}
          initial="hidden"
          whileInView="visible"
        >
          <motion.div variants={fadeInVariant}>
            <TopArticle article={blog.articles[0]} />
          </motion.div>
          <div>
            <motion.h2 variants={fadeInVariant} className="recent_post">
              Recent posts
            </motion.h2>
            <ArticlesGrid articles={blog.articles} />
          </div>
        </motion.div>
      </div>
    </PageContainer>
  );
}

export default BlogHomepage;
