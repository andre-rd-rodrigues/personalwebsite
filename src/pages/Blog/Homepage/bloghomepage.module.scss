@import "assets/styles/mixins";

.container {
  @include blog_settings;
}

.blogHomepage {
  p[class="article_description"] {
    text-align: justify;
  }
}
