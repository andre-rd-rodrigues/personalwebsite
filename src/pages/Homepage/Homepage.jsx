import React from "react";
import { PopupButton } from "@typeform/embed-react";
import { analyticsClickEvent } from "analytics/utils";
import {
  fadeInVariant,
  homepageDelayVariant
} from "assets/motion/motionVariants";
import AnimatedHeading from "components/AnimatedHeading/AnimatedHeading";
import AppLink from "components/AppLink/AppLink";
import GalleryBackgroundSlider from "components/Gallery/GalleryBackgroundSlider";
import { homepage_gallery } from "data";
import { motion } from "framer-motion";
import { useTranslation } from "react-i18next";

import styles from "./homepage.module.scss";

const Homepage = () => {
  const { t } = useTranslation();

  return (
    <div className={styles.homepage}>
      <GalleryBackgroundSlider images={homepage_gallery}>
        <div className={styles.title}>
          <AnimatedHeading centerOnSmallScreens>
            <h1>Andr√© Rodrigo</h1>
          </AnimatedHeading>
          <motion.div
            variants={homepageDelayVariant}
            initial="hidden"
            animate="visible"
          >
            <motion.p variants={fadeInVariant}>
              {t("homepage.subtitle")}
            </motion.p>
            <motion.div
              variants={fadeInVariant}
              onClick={() =>
                analyticsClickEvent(
                  "Portfolio Link",
                  "Portfolio - clicked link"
                )
              }
            >
              <AppLink to="/portfolio" label={t("homepage.cta")} scrollTop />|
              <PopupButton
                id="wTr5ba0e"
                size={60}
                className={styles.formButton}
              >
                {t("homepage.form.button")}
              </PopupButton>
            </motion.div>
          </motion.div>
        </div>
      </GalleryBackgroundSlider>
    </div>
  );
};

export default Homepage;
