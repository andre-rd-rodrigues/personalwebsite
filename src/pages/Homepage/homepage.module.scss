@import "assets/styles/mixins";

.homepage {
  background-color: $dark;
}
.title {
  color: white;
  h1 {
    font-size: var(--fs-xxl);
  }
  p {
    font-size: var(--fs-m);
    margin: 24px;
  }
  a {
    font-size: var(--fs-m);
  }
}

.formButton {
  position: relative;
  text-transform: uppercase;
  letter-spacing: 3px;
  color: white;
  font-weight: 400;
  border: none;
  background: none;
  @include hover_underline($white);
  &:hover {
    letter-spacing: 6px;
  }
}
