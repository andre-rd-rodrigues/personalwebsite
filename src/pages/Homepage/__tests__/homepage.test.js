import { fireEvent, screen } from "@testing-library/react";
import { render } from "utils/tests/test-utils";

import Homepage from "../Homepage";

describe("Contact page", () => {
  it("renders page correctly", () => {
    const { asFragment } = render(<Homepage />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("cta should redirect to correct page", () => {
    render(<Homepage />);

    const linkButton = screen.getByText(/portfolio/i);

    fireEvent.click(linkButton);

    expect(window.location.href).toBe("http://localhost/portfolio");
  });
});
