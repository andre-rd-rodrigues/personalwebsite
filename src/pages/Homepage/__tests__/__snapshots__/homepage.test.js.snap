// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact page renders page correctly 1`] = `
<DocumentFragment>
  <div
    class="homepage"
  >
    <div
      class="layer"
      style="opacity: 0;"
    >
      <div
        class="container"
      >
        <div
          class="background"
        >
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
        </div>
        <div
          class="content"
        >
          <div
            class="title"
          >
            <div
              class="container undefined"
              style="justify-content: center;"
            >
              <span
                style="white-space: nowrap;"
              >
                <span
                  style="overflow: hidden; display: inline-block;"
                >
                  <h1
                    style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
                  >
                    A
                  </h1>
                </span>
                <span
                  style="overflow: hidden; display: inline-block;"
                >
                  <h1
                    style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
                  >
                    n
                  </h1>
                </span>
                <span
                  style="overflow: hidden; display: inline-block;"
                >
                  <h1
                    style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
                  >
                    d
                  </h1>
                </span>
                <span
                  style="overflow: hidden; display: inline-block;"
                >
                  <h1
                    style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
                  >
                    r
                  </h1>
                </span>
                <span
                  style="overflow: hidden; display: inline-block;"
                >
                  <h1
                    style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
                  >
                    é
                  </h1>
                </span>
                <span
                  style="overflow: hidden; display: inline-block;"
                >
                  <h1
                    style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
                  >
                     
                  </h1>
                </span>
              </span>
              <span
                style="white-space: nowrap;"
              >
                <span
                  style="overflow: hidden; display: inline-block;"
                >
                  <h1
                    style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
                  >
                    R
                  </h1>
                </span>
                <span
                  style="overflow: hidden; display: inline-block;"
                >
                  <h1
                    style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
                  >
                    o
                  </h1>
                </span>
                <span
                  style="overflow: hidden; display: inline-block;"
                >
                  <h1
                    style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
                  >
                    d
                  </h1>
                </span>
                <span
                  style="overflow: hidden; display: inline-block;"
                >
                  <h1
                    style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
                  >
                    r
                  </h1>
                </span>
                <span
                  style="overflow: hidden; display: inline-block;"
                >
                  <h1
                    style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
                  >
                    i
                  </h1>
                </span>
                <span
                  style="overflow: hidden; display: inline-block;"
                >
                  <h1
                    style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
                  >
                    g
                  </h1>
                </span>
                <span
                  style="overflow: hidden; display: inline-block;"
                >
                  <h1
                    style="display: inline-block; opacity: 0; transform: translateY(200%) translateZ(0);"
                  >
                    o
                  </h1>
                </span>
              </span>
            </div>
            <div
              style="opacity: 0;"
            >
              <p
                style="opacity: 0;"
              >
                Production of websites, apps and digital presence
              </p>
              <div
                style="opacity: 0;"
              >
                <a
                  class="appLink"
                  href="/portfolio"
                >
                  Portfolio
                </a>
                |
                <style>
                  @keyframes spin{to{transform:rotate(360deg)}}.tf-v1-popup{position:fixed;top:0;left:0;width:100%;height:100%;background:rgba(0,0,0,.75);transition:opacity .25s ease-in-out;z-index:10001;display:flex;align-items:center;justify-content:center}.tf-v1-popup .tf-v1-iframe-wrapper{position:relative;transition:opacity .25s ease-in-out;min-width:360px;min-height:360px}.tf-v1-popup .tf-v1-iframe-wrapper iframe{width:100%;height:100%;border:none;overflow:hidden;border-radius:8px}.tf-v1-popup .tf-v1-close{position:absolute;font-size:32px;line-height:24px;width:24px;height:24px;text-align:center;cursor:pointer;opacity:.75;transition:opacity .25s ease-in-out;text-decoration:none;color:#000;top:-34px;right:0}.tf-v1-popup .tf-v1-close:hover{opacity:1}@media(min-width: 481px){.tf-v1-popup .tf-v1-close{color:#fff !important}}.tf-v1-popup .tf-v1-spinner{border:3px solid #aaa;font-size:40px;width:1em;height:1em;border-radius:.5em;box-sizing:border-box;animation:spin 1s linear infinite;border-top-color:#fff;position:absolute;top:50%;left:50%;margin:-20px 0 0 -20px}@media(max-width: 480px){.tf-v1-popup{width:100% !important;height:100% !important;width:100vw !important;height:100vh !important;max-height:stretch}.tf-v1-popup .tf-v1-iframe-wrapper{position:relative;transition:opacity .25s ease-in-out;min-width:100%;min-height:100%}.tf-v1-popup .tf-v1-iframe-wrapper iframe{border-radius:0}.tf-v1-popup .tf-v1-close{position:absolute;font-size:32px;line-height:24px;width:24px;height:24px;text-align:center;cursor:pointer;opacity:.75;transition:opacity .25s ease-in-out;text-decoration:none;color:#000;top:6px;right:8px}.tf-v1-popup .tf-v1-close:hover{opacity:1}}@media(max-width: 480px)and (min-width: 481px){.tf-v1-popup .tf-v1-close{color:#fff !important}}
                </style>
                <button
                  class="formButton"
                  data-testid="tf-v1-popup"
                >
                  Work with me
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
